### PR TITLE
feat (applications): add the ability to create as a task

### LINKED ISSUE #629: clever create --task
To create a task a user need to know it exists and add manually `CC_TASK=true`. It could be great to create en app as a task with a `--task` flag detailed in the helper. It can be done through the API with the parameter `instanceLifetime` set to `TASK`

### PR DESCRIPTION
**Pull request by davlgd**
Fix #629

### CODE REVIEW

**Comment by davlgd** [comment]
Branch is rebased, I added JSON format support. For now, in JSON mode the CC_RUN_COMMAND won't be added at app creation (as it show a human message). But as env set at creation will be available through API soon (I have to check status), I propose to clarify this in documentation for now, and add JSON support in a future PR.

**Comment by davlgd** [comment]
As API now supports env vars set at application creation, I use it. It allows me to set `CC_RUN_COMMAND` also in JSON mode (with no `human` message). We don't have a confirmation of the env var set in response, I'll raise it as an API issue to know how we can get such an information and expose a confirmation to user later.

**Comment by aurrelhebert** [comment]
I did some QA: 
 - :heavy_check_mark:  create a task (`clever create -t node --task` 
 - ❌ when `--task` is empty 
 - ❌ when I tried to create a node app
 
 For the create I go the following results: 
```sh
# When setting an empty `--task`
clever create empty -t node --task
Your application has been successfully created as a task!
ID: app_ID
Name: empty
```
However the created doesn't have a `CC_RUN_COMMAND` set. Maybe we should output an error for this case?

 When creating a node app I got: 
```sh
clever create app -t node
[ERROR] Cannot read properties of null (reading 'length')
```

**Comment by aurrelhebert** [comment]
Thanks for the patch. It's maybe that have wronly understood what is application task :) 

An application task with no `CC_RUN_COMMAND` set could exist? Otherwise I would maybe prefer an error than an empty CC_RUN_COMMAND.

**Comment by davlgd** [comment]
`CC_RUN_COMMAND` is mandatory for Tasks, but I wouldn't force the user to set it within the app creation process (and I wasn't sure it will be a way to create a task in final implementation)

**Review by davlgd — COMMENTED** [review]
LGTM after some text fixes

File: bin/clever.js [hunk 1]
```diff
@@ -360,6 +360,12 @@ function run () {
       metavar: 'token',
       description: 'Directly give an existing token',
     }),
+    taskCommand: cliparse.option('task', {
+      description: 'The application launch as a task executing the given command, then stopped',
+      aliases: ['T'],
+      parser: Parsers.nonEmptyString,
+      metavar: 'task',
```
[the remark this answers is not in the document]
Comment (hsablonniere) [reply bin/clever.js:367]: I wasn't sure what's the use of metavar, do you know/remember?

File: src/parsers.js [hunk 2]
```diff
@@ -94,6 +94,13 @@ function integer (string) {
   return cliparse.parsers.success(integer);
 }
 
+function nonEmptyString (string) {
+  if (typeof string !== 'string' || string === '') {
+    return cliparse.parsers.error('Invalid string, it should not be empty');
```
[the remark this answers is not in the document]
Comment (hsablonniere) [reply src/parsers.js:99]: I would agree on this but all other parser error messages start with an uppercase letter.

I propose to merge with the uppercase and you can create an issue so we can update it everywhere with a lowercase.

File: bin/clever.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (davlgd) [reply bin/clever.js:367]: It's how is show in help, for example :
image

[the remark this answers is not in the document]
Comment (hsablonniere) [reply bin/clever.js:367]: Indeed, I 100% forgot about this. I'll apply :+1:

**Review by hsablonniere — APPROVED** [review]

**Pull request merged**